src/google/iam.js

```diff
@@ -25,7 +42,7 @@ async function getServiceAccount(project, name, auth) {
       timeout: 2000,
```

Comment:
The options passed to/returned by the Google API method [auth.authorizeRequest](https://googleapis.dev/nodejs/google-auth-library/5.2.2/classes/GoogleAuth.html#authorizeRequest) are expected to be `Axios` or `Request` options. It's not obvious if e.g. the `timeout` option is actually used during the async call to `authorizeRequest`.  This is confusing at best.

I suggest to use [auth.getRequestHeaders](https://googleapis.dev/nodejs/google-auth-library/5.2.2/classes/GoogleAuth.html#getRequestHeaders) instead and build Fetch-specific options.